run_robot_test.sh

```diff
@@ -71,29 +75,135 @@ while [ "$#" -gt 0 ]; do
+            t) # Specify test case to run
+                TEST_CASE_FILE=$TEST_CASE_LOCATION/$OPTARG
+            ;;
+            v) # Override/Add global variables specified in the test variables file
+                TEST_VARIABLES="${TEST_VARIABLES} --variable $OPTARG"
+            ;;
+            *)
+                echo "Invalid Input"
+                disp_usage $OPTSTR
+                exit 1
+            ;;
+        esac
+    done
+
+    shift $((OPTIND -1))
+
+    # ------ Validate Inputs --------
+    if [[ ! -f "${TEST_VARIABLES_FILE}" ]]; then
+        echo "---- Robot Framework test variable file (test-variables.yml) is missing"
+        exit 1
+    fi
+}
+
+
+disp_usage() {
+    echo
+    echo "Usage:    $0 [$1]"
+    echo "Possible inputs: -d -e -f [-h] -i -r [-s] -t -v"
+    echo "Specified Inputs: $SPECIFIED_INPUTS"
+    echo "Explanation of inputs "
+    echo "      -d <TEST_ARTIFACT_DIR>: Specify directory to store artifacts and reports from each test run"
+    echo "      -e <TEST_EXCLUDE_TAG>: Specify excluded tags for Robot"
+    echo "      -f <TEST_VARIABLES_FILE>: Specify the test variable file"
+    echo "      -h: Display Usage and exit"
+    echo "      -i <TEST_INCLUDE_TAG>: Specify included tags for Robot"
+    echo "      -r <EXTRA_ROBOT_ARGS>: Additional arguments to pass to the robot cli"
+    echo "      -s: Skip the installation of required Python Libraries during the execution of this script"
+    echo "      -t <TEST_CASE_FILE>: Specify test case to run, should be present under Tests directory"
+    echo "      -v <>: Override/Add global variables specified in the test variables file"
+}
+
+
+pip_install() {
+    # This is optional so we are not creating/updating the virtualenv everytime we run a test
+    echo "--------- Installing required Python Libraries ----------"
+    VENV_ROOT=${PWD}/venv
+    # setup virtualenv
+    python3 -m venv ${VENV_ROOT}
+    source ${VENV_ROOT}/bin/activate
+
+    ${VENV_ROOT}/bin/pip install --upgrade pip
+    ${VENV_ROOT}/bin/pip install -r requirements.txt
+}
+
+
+create_artifact_dir() {
+    # Create a unique directory to store the output for current test run
+    if [[ ! -d "${TEST_ARTIFACT_DIR}" ]]; then
+      mkdir ${TEST_ARTIFACT_DIR}
+    fi
+    case "$(uname -s)" in
+        Darwin)
+            TEST_ARTIFACT_DIR=$(mktemp -d  ${TEST_ARTIFACT_DIR} -t ${TEST_ARTIFACT_DIR}/rhoda-ci-$(date +%Y-%m-%d-%H-%M)-XXXXXXXXXX)
+             ;;
+        Linux)
+            TEST_ARTIFACT_DIR=$(mktemp -d -p ${TEST_ARTIFACT_DIR} -t rhoda-ci-$(date +%Y-%m-%d-%H-%M)-XXXXXXXXXX)
+            ;;
+    esac
+}
 
-if [[ ! -f "${TEST_VARIABLES_FILE}" ]]; then
-  echo "Robot Framework test variable file (test-variables.yml) is missing"
-  exit 1
+
+echo "------------- STARTING ${0}"
+handle_inputs "$@"
+
+if [[ $SKIP_PIP_INSTALL -eq 0 ]]; then
+    pip_install
+else
+    echo "---- Python Libraries installation skipped"
+    echo "---- To not skip the installation, remove the -s flag"
```

Comment:
Let's avoid the `else` part and in case of debug we can `set +x` during execution.

Reply:
Done